src/sidebar/components/test/thread-list-omega-test.js

```diff
@@ -0,0 +1,280 @@
+
+      assert.calledOnce(fakeStore.clearSelection);
+    });
+
+    it('does not clear the selection in the store if new annotation is a highlight', () => {
+      fakeMetadata.isHighlight.returns(true);
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.notCalled(fakeStore.clearSelection);
+    });
+
+    it('does not clear the selection in the store if new annotation is a reply', () => {
+      fakeMetadata.isReply.returns(true);
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.notCalled(fakeStore.clearSelection);
+    });
+  });
+
+  context('active scroll to an annotation thread', () => {
+    let stubbedDocument;
+    let stubbedFakeScrollContainer;
+    let fakeScrollTop;
+
+    beforeEach(() => {
+      fakeScrollTop = sinon.stub();
+      stubbedFakeScrollContainer = sinon
+        .stub(fakeScrollContainer, 'scrollTop')
+        .set(fakeScrollTop);
+      stubbedDocument = sinon
+        .stub(document, 'querySelector')
+        .returns(fakeScrollContainer);
+    });
+
+    afterEach(() => {
+      stubbedFakeScrollContainer.restore();
+      stubbedDocument.restore();
+    });
+
+    it('should do nothing if there is no active annotation thread to scroll to', () => {
+      createComponent();
+
+      assert.notCalled(fakeScrollTop);
+    });
+
+    it('should do nothing if the annotation thread to scroll to is not in DOM', () => {
+      createComponent();
+
+      act(() => {
+        fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED](
+          {},
+          { $tag: 'nonexistent' }
+        );
+      });
+
+      assert.notCalled(fakeScrollTop);
+    });
+
+    it('should set the scroll container `scrollTop` to derived position of thread', () => {
+      createComponent();
+
+      act(() => {
+        fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED](
+          {},
+          fakeTopThread.children[3].annotation
+        );
+      });
+
+      // The third thread in a collection of threads at default height (200)
+      // should be at 600px. This setting of `scrollTop` is the only externally-
+      // observable thing that happens here...
+      assert.calledWith(fakeScrollTop, 600);
+    });
+  });
+
+  describe('scroll and resize event handling', () => {
+    const debouncedFn = sinon.stub();
+
+    beforeEach(() => {
+      fakeDebounce.returns(debouncedFn);
+    });
+
+    it('updates scroll position and window height for recalculation on container scroll', () => {
+      createComponent();
+      document
+        .querySelector('.js-thread-list-scroll-root')
+        .dispatchEvent(new Event('scroll'));
+
+      assert.calledOnce(debouncedFn);
+    });
+
+    it('updates scroll position and window height for recalculation on window resize', () => {
+      createComponent();
+      window.dispatchEvent(new Event('resize'));
+
+      // TODO
```

Comment:
Triggering a `resize` event on the `window` here causes the callback to fire twice—is it possible that a `scroll` event is getting triggered on the scroll container? Or that a window resize event somehow causes recalculation to happen that then triggers another `resize` or `scroll` event? Need a help tracking this down.

Reply:
Not sure from just looking at the code. Add some logging in to see what is triggering the function being called?

Reply:
Dang it, this was a testing setup failure; the `debouncedFn` sinon stub was getting reused in the second test in this block instead of reset.